### PR TITLE
Set exception handler feature in developer exception page

### PR DESCRIPTION
**Pull request by bjornen77**
When using the developer exception page, the exception handler
feature is now set before invoking the problem details service.
This makes it possible to get the original exception when using
the problem details service if wanted.

Fixes# https://github.com/dotnet/aspnetcore/issues/47060

### CODE REVIEW

File: src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddlewareImpl.cs [hunk 1]
```diff
@@ -88,6 +88,22 @@ private static ExtensionsExceptionJsonContext CreateSerializationContext(JsonOpt
         return new ExtensionsExceptionJsonContext(new JsonSerializerOptions(jsonOptions.SerializerOptions));
     }
 
+    private static void SetExceptionHandlerFeatures(ErrorContext errorContext)
+    {
+        var httpContext = errorContext.HttpContext;
+
+        ExceptionHandlerFeature exceptionHandlerFeature = new()
```
Comment (JamesNK) [inline src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddlewareImpl.cs:95]:
```suggestion
        var exceptionHandlerFeature = new ExceptionHandlerFeature()
```

**Comment by JamesNK** [comment]
This is ok, but have we considered adding the exception to the problem details context?

```csharp
public class ProblemDetailsContext
{
    public Exception? Exception { get; init; }
}
```

When the problem details context is written from one of the exception handler middleware, the exception would be set on the context.

Getting the exception from a feature is a semi-advanced concept. Having it available on the problem details context would be more straightforward.

cc @captainsafia @mitchdenny @BrennanConroy @amcasey

File: src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddlewareImpl.cs (hunk 1 above)
Comment (bjornen77) [reply src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddlewareImpl.cs]: Thanks for your suggestion, fixed.

**Comment by bjornen77** [comment]
I do not know if adding the exception to the context has been discussed, but I think it would be a good thing to do. Should we keep this change just to make the different exception handlers behave the same?

**Comment by captainsafia** [comment]
> Getting the exception from a feature is a semi-advanced concept. Having it available on the problem details context would be more straightforward.

FWIW, I don't think getting it from the feature is that advanced. If anything, it has a bit of an edge over the `ProblemDetailsContext` since it relies on an API that's been around for longer. That being said, I'm not opposed to adding an `Exception` to the `ProblemDetails`.

> Should we keep this change just to make the different exception handlers behave the same?

I think we should go with this change because adding the `Exception` to the `ProblemDetails` context would be additive. We should set the feature in the DEP even if it was set in the exception so it is accessible to people not using the `ProblemDetailsService`.

> I do not know if adding the exception to the context has been discussed, but I think it would be a good thing to do.

Would you be interested in submitting an API proposal for this via [the template](https://github.com/dotnet/aspnetcore/issues/new?assignees=&labels=api-suggestion&template=30_api_proposal.md&title=)?

**Review by captainsafia — APPROVED** [review]
LGTM given that I believe having the feature on the `ProblemDetailsContext` should happen _in addition_ to this change.

**Comment by bjornen77** [comment]
> Would you be interested in submitting an API proposal for this via [the template](https://github.com/dotnet/aspnetcore/issues/new?assignees=&labels=api-suggestion&template=30_api_proposal.md&title=)?

Sure, no problem!

**Pull request merged**